File: leetcode/1802-number-of-students-unable-to-eat-lunch/solution.cpp
class Solution {
public:
    int countStudents(vector<int>& students, vector<int>& sandwiches) {
        int cnt0 = 0;
        int cnt1 = 0;
        int s = students.size();

        for(int i = 0; i < s; i++) {
            if (students[i] == 0) {
                cnt0++;
            } 
            else {
                cnt1++;
            }
        }
        for(int i = 0; i < s; i++) {
            if ((sandwiches[i] == 0 and cnt0 == 0) or (sandwiches[i] == 1 and cnt1 == 0)){
                return s - i;
            }
            if (sandwiches[i] == 0) {
                cnt0--;
            }
            else {
                cnt1--;
            }
        }
        return 0;
        

    }
};

File: leetcode/1802-number-of-students-unable-to-eat-lunch/solution.py
class Solution:
    def countStudents(self, students: List[int], sandwiches: List[int]) -> int:
        size = len(students)
        cnt0 = 0
        cnt1 = 0
        
        for s in students:
            if s == 0:
                cnt0 += 1 
            else:
                cnt1 += 1

        for i, san in enumerate(sandwiches): 
            if san == 0 and cnt0 == 0:
                return size - i
            if san == 1 and cnt1 == 0:
                return size - i
            
            if san == 0:
                cnt0 -=1
            else:
                cnt1 -=1
        
        return 0

